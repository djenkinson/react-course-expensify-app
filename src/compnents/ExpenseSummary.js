import React from 'react';
import { connect} from 'react-redux';
import selectExpenses from '../selectors/expenses';
import selectSummary from '../selectors/expenses-total';
import numeral from 'numeral';

export const ExpenseSummary = (props) => (
    <div>
        {
            props.expensesCount === 1 ? (
                <p>Viewing 1 expense totalling {'£' + numeral(props.expensesTotal/100).format('0,0.00')}</p>
            ) : (
                props.expensesCount > 0 ? (
                    <p>Viewing {props.expensesCount} expenses totalling {'£' + numeral(props.expensesTotal/100).format('0,0.00')}</p>
                ) : (
                    <p></p>
                )
            )
        }
    </div>
);



const mapStateToProps = (state) => {
    const expenses = selectExpenses(state.expenses, state.filters)
    return {
        expensesCount: expenses.length,
        expensesTotal: selectSummary(expenses)
    }
};

export default connect(mapStateToProps)(ExpenseSummary);