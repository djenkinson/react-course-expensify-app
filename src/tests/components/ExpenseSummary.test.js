import React from 'react';
import { shallow } from 'enzyme';
import { ExpenseSummary } from '../../compnents/ExpenseSummary';
import expenses from '../fixtures/expenses';

test('should render ExpenseSummary with 1 expense', () => {
    const wrapper = shallow(<ExpenseSummary expensesCount={1} expensesTotal={9434}/>);
    expect(wrapper).toMatchSnapshot();
});

test('should render ExpenseSummary with 3 expenses', () => {
    const wrapper = shallow(<ExpenseSummary expensesCount={2} expensesTotal={9434}/>);
    expect(wrapper).toMatchSnapshot();
});