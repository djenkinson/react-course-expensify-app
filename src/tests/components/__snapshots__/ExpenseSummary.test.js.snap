// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseSummary with 1 expense 1`] = `
<div>
  <p>
    Viewing 1 expense totalling 
    £94.34
  </p>
</div>
`;

exports[`should render ExpenseSummary with 3 expenses 1`] = `
<div>
  <p>
    Viewing 
    2
     expenses totalling 
    £94.34
  </p>
</div>
`;
